lib/main_common.pm

```diff
@@ -1600,44 +1601,86 @@ sub load_filesystem_tests {
     loadtest 'console/snapper_used_space' if (is_sle('15-SP1+') || (is_opensuse && !is_leap('<15.1')));
 }
 
+sub get_wicked_tests {
+    if (check_var('WICKED', 'basic')) {
+        return qw(
+          t01_basic
+          t02_static_addresses_legacy
+          t03_static_addresses_xml
+          t04_dynamic_addresses_legacy
+          t05_dynamic_addresses_xml
+          t06_static_routes_legacy
+          t07_static_routes_xml
+        );
+    }
+    elsif (check_var('WICKED', 'advanced')) {
+        return qw(
+          t01_gre_tunnel_legacy
+          t02_gre_tunnel_xml
+          t03_sit_tunnel_legacy
+          t04_sit_tunnel_xml
+          t05_ipip_tunnel_legacy
+          t06_ipip_tunnel_xml
+          t07_tun_interface_legacy
+          t08_tun_interface_xml
+          t09_tap_interface_legacy
+          t10_tap_interface_xml
+          t11_bridge_interface_legacy
+          t12_bridge_interface_xml
+        );
+    }
+    elsif (check_var('WICKED', 'startandstop')) {
+        return qw(
+          t01_standalone_card_ifdown_ifreload
+          t02_bridge_ifreload
+          t03_bridge_ifup_ifreload
+          t04_bridge_ifup_remove_all_config_ifreload
+          t05_bridge_ifup_remove_one_config_ifreload
+          t06_bridge_ifdown_create_new_config_ifreload_ifdown_ifup
+          t07_bridge_ifdown_remove_one_config_ifreload_ifdown_ifup
+          t08_sit_tunnel_ifdown
+          t09_openvpn_tunnel_ifdown
+          t10_vlan_ifup_all_ifdown_one_card
+        );
+    }
+    else {
+        die 'Unhandled WICKED test selection: ' . get_var('WICKED');
+    }
+}
+
+sub load_wicked_barrier {
+    if (check_var('WICKED', 'advanced') ||
+        check_var('WICKED', 'startandstop')) {
+        my $args         = OpenQA::Test::RunArgs->new();
+        my @wicked_tests = get_wicked_tests();
+        $args->{wicked_tests} = \@wicked_tests;
+        loadtest('wicked/barrier_init', run_args => $args);
+    }
+}
+
 sub load_wicked_tests {
     loadtest 'wicked/before_test';
+    my $folder;
```

Comment:
Not very important but "folder" is rather a Windows concept, I'd prefer to use $directory or $dir